File: day-02-2/java/src/test/java/com/adventofcode/AnswerTest.java
package com.adventofcode;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.mockito.MockedStatic;
import org.mockito.Mockito;

class AnswerTest {

    public final static String EXAMPLE = "forward 5\ndown 5\nforward 8\nup 3\ndown 8\nforward 2";

    @Test
    public void testSolution() {
        try (MockedStatic<Answer> utilities = Mockito.mockStatic(Answer.class)) {
            utilities.when(Answer::readInputFile).thenReturn(EXAMPLE);
            assertEquals(900, new Answer().solution());
        }
    }

}

File: day-02-2/ruby/answer_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'answer'

class AnswerTest < Minitest::Test
  EXAMPLE = "forward 5\n" \
            "down 5\n"    \
            "forward 8\n" \
            "up 3\n"      \
            "down 8\n"    \
            'forward 2'

  def test_main
    Answer.stub(:read_input_file, EXAMPLE) do
      assert_equal 900, Answer.new.main
    end
  end
end
